messaging-client-databus/src/main/java/com/inmobi/databus/partition/PartitionStreamReader.java

```diff
@@ -45,4 +45,6 @@
   MessageCheckpoint getMessageCheckpoint();
 
   MessageCheckpoint buildStartPartitionCheckpoints();
+
+  Long getReaderBackLog() throws IOException, Exception;
```

Comment:
Exception already subsumes IOException. However we should try to avoid throwing Exception and instead throw required subclasses as much as possible.